src/compas_fab/robots/robot.py

```diff
@@ -302,506 +464,696 @@ def get_RCF(self, group=None):
-        joint_positions = self.scale_joint_values(joint_positions, self.scale_factor)
-        response.start_configuration = Configuration(joint_positions, self.get_configurable_joint_types())
+        joint_positions = response.solution.joint_state.position
+        joint_positions = self._scale_joint_values(
+            joint_positions, self.scale_factor)
+        # full configuration # TODO group config
+        configuration = Configuration(
+            joint_positions, self.get_configurable_joint_types())
 
-        return response
+        return configuration
 
-    def motion_plan_goal_frame(self, frame_WCF, start_configuration,
-                               tolerance_position, tolerance_angle,
-                               group=None, path_constraints=None,
-                               trajectory_constraints=None, planner_id='RRT',
-                               num_planning_attempts=8, allowed_planning_time=2.,
-                               max_velocity_scaling_factor=1.,
-                               max_acceleration_scaling_factor=1.,
-                               attached_collision_object=None):
-        """Calculates a motion from start_configuration to frame_WCF.
+    def forward_kinematics(self, configuration, group=None):
+        """Calculate the robot's forward kinematic.
 
         Parameters
         ----------
-            frame_WCF (:class:`Frame`): The goal frame.
-            start_configuration (:class:`Configuration`, optional): The robot's
-                configuration at the starting position.
-            tolerance_position (float): the allowed tolerance to the frame's
-                position. (Defined in the robot's units)
-            tolerance_angle (float): the allowed tolerance to the frame's
-                orientation in radians.
-            group (str, optional): The planning group used for calculation.
-                Defaults to the robot's main planning group.
+        configuration : :class:`compas_fab.robots.Configuration`
+            The configuration to calculate the forward kinematic for.
+        group : str, optional 
+            The planning group used for the calculation. Defaults to the robot's
+            main planning group.
 
         Examples
         --------
+        >>> from compas_fab.backends import RosClient
+        >>> from compas_fab.backends import RosError
+        >>> from compas_fab.robots import Configuration
+        >>> from compas_fab.robots.ur5 import Robot
+        >>> client = RosClient()
+        >>> client.run()
+        >>> robot = Robot(client)
+        >>> configuration = Configuration.from_revolute_values([-2.238, -1.153, -2.174, 0.185, 0.667, 0.000])
+        >>> group = robot.main_group_name
+        >>> response = robot.forward_kinematics(configuration, group)
+        >>> client.close()
+        >>> client.terminate()
         """
+        # TODO implement with no service, only geometry transformations
```

Comment:
A purely geometrical FK could perhaps be a separate method (either in this class or as a function on its own), but I think using the client backend's FK service should always be the default option.